cmake/try_compile_clang_cuda.cmake

```diff
@@ -0,0 +1,45 @@
+# Tries CUDA compilation using clang
+#
+# Usage of this function:
+#
+#  try_compile_clang_cuda(CUDA_ARCH)
+#
+#  Parameters:
+#    - CUDA_ARCH: architecture used for testing compilation
+#                 (doesn't have to match the architecture of an available GPU, no code is executed)
+#
+# This function defines:
+#  GT_CLANG_CUDA_WORKS         CUDA test file was successfully compiled with clang
+#
+function(try_compile_clang_cuda CUDA_ARCH)
```

Comment:
Can you pass the name of the output variable as a parameter? (The same also for hip detection function.)